specs/mgt-search-suggestion.md

```diff
@@ -0,0 +1,161 @@
+
+The following examples shows how to use the `error` template.
+
+```html
+<mgt-search-suggestion>
+    <template data-type="search-suggestion-texts">
+        <div  data-for="text in texts">
+        {{text.text}}
+        </div>
+    </template>
+</mgt-search-suggestion>
+
+```
+
+## Proposed Solution
+
+### Example 1: basic usage without any callback function
+```<mgt-search-suggestion></mgt-search-suggestion>```
+
+### Example 2: Developer provides a site-id and item-id
+```<mgt-search-suggestion id="search-suggestion"> </mgt-search-suggestion>```
+```
+#User can get this component then bind callback action to this component 
+
+function onClickCallback(suggestionValue) {
+    console.log('suggestion value:', suggestionValue);
+    var searchValue = getSuggestionValue(suggestionValue);
+    window.location.assign('https://www.bing.com/search?q=' + searchValue);
+}
+
+function onEnterKeyPressCallback(originalValue, selectedSuggestionValue) {
+    console.log('original value:', originalValue);
+    console.log('suggestion value:', selectedSuggestionValue);
+    var searchValue = getSuggestionValue(selectedSuggestionValue);
+    window.location.assign('https://www.bing.com/search?q=' + searchValue);
+}
+
+function getSuggestionValue(suggestionValue) {
+    var searchValue = '';
+    if (suggestionValue.entity == 'File') {
+        searchValue = suggestionValue.name;
+    } else if (suggestionValue.entity == 'Text') {
+        searchValue = suggestionValue.text;
+    } else if (suggestionValue.entity == 'People') {
+        searchValue = suggestionValue.displayName;
+    }
+    return searchValue;
+}
+
+var obj = document.getElementById('search-suggestion');
+obj.onClickCallback = onClickCallback;
+obj.onEnterKeyPressCallback = onEnterKeyPressCallback;
+
+```
+
+### Example 3: Developer can select entity types
+
+```<mgt-search-suggestion selected-entity-types="file, text, people"></mgt-search-suggestion>```
+
+```<mgt-search-suggestion selected-entity-types="file, people"></mgt-search-suggestion>```
+
+```<mgt-search-suggestion selected-entity-types="people, text"></mgt-search-suggestion>```
+
+## Example 4: Developer can set each entity type's suggestion count
+
+```<mgt-search-suggestion max-text-suggestion-count="3" max-file-suggestion-count="2" max-people-suggestion-count="3"></mgt-search-suggestion>```
+
+## Attributes and Properties
+
+| Attribute | Property | Description |
+| --------- | -------- | ----------- |
+| `max-text-suggestion-count` | `maxTextSuggestionCount` | The max suggestion count for text. |
+| `max-people-suggestion-count` | `maxPeopleSuggestionCount` | The max suggestion count for people. |
+| `max-file-suggestion-count` | `maxFileSuggestionCount` | The max suggestion count for file. |
+| `selected-entity-types` | `selectedEntityTypes` | Suggestion entity types, free combination of text/people/file, use ',' to do the segmentation |
```

Comment:
Would `suggested-entity-types` work better here?

```suggestion
| `selected-entity-types` | `selectedEntityTypes` | Suggestion entity types, free combination of text/people/file, use ',' to do the segmentation |
```

Reply:
I think they are same, select is an action for me, developer select some entity types from text/people/file.

Reply:
I made big  changes here, so for current now, the usage should like below.
(Suppose suggestion API support sample1 entity type)

 <mgt-search entity-types="query-2, file-3, people-1, sample1"></mgt-search>
The rendering order is:  query , file, people, sample1
The suggestion amount limitation is: 2, 3, 1, 3 ( 3 is default value )

Reply:
Another possible option:

To show max 3 suggestions for each entity type:

 `<mgt-search entity-types="query, file, people" max-suggestions="3"></mgt-search>`

To set different max suggestion counts for each entity type:

 `<mgt-search entity-types="query, file, people" max-suggestions="2,3,1"></mgt-search>`

@nmetulev @beth-panx any input here?

Reply:
I just had the same suggestion, I agree with @elisenyang - try to keep it simple at first

Reply:
sure, let's use max-suggestions="3"